web/src/main/kotlin/com/example/ajaxproject/repository/impl/GroupChatRoomRepositoryImpl.kt

```diff
@@ -3,39 +3,58 @@ package com.example.ajaxproject.repository.impl
 import com.example.ajaxproject.model.GroupChatRoom
 import com.example.ajaxproject.model.User
 import com.example.ajaxproject.repository.GroupChatRoomRepository
-import org.springframework.data.mongodb.core.MongoTemplate
+import org.springframework.data.mongodb.core.ReactiveMongoTemplate
 import org.springframework.data.mongodb.core.query.Criteria
 import org.springframework.data.mongodb.core.query.Query
 import org.springframework.data.mongodb.core.query.Update
 import org.springframework.stereotype.Repository
+import reactor.core.publisher.Flux
+import reactor.core.publisher.Mono
 
 @Repository
 class GroupChatRoomRepositoryImpl(
-    private val mongoTemplate: MongoTemplate
+    private val reactiveMongoTemplate: ReactiveMongoTemplate
 ) : GroupChatRoomRepository {
 
-    override fun findChatRoom(chatId: String): GroupChatRoom {
+    override fun findChatRoom(chatId: String): Mono<GroupChatRoom> {
         val chatQuery = Query.query(Criteria.where("_id").`is`(chatId))
-        validateAndCleanChatMembers(chatId)
-        return mongoTemplate.findOne(chatQuery, GroupChatRoom::class.java)!!
+        return reactiveMongoTemplate.findOne(chatQuery, GroupChatRoom::class.java)
+            .flatMap { chatRoom ->
+                validateAndCleanChatMembers(chatId)
```

Comment:
why `findChatRoom` does `validateAndCleanChatMembers`?

Find something should only **find**. 

You put some business logic at repository layer that inappropriate.

Reply:
I fixed it